lib/swagger/diff/specification.rb

```diff
@@ -44,36 +44,44 @@ def merge_refs!(h1, h2)
       end
 
       def params_or_nil(endpoint)
-        endpoint && endpoint.parameters || nil
+        endpoint && endpoint['parameters'] || nil
       end
 
       def parse_swagger(swagger)
         if swagger.is_a? Hash
-          Swagger.build(swagger)
-        elsif File.exist?(swagger)
-          Swagger.load(swagger)
+          swagger
         else
-          swagger = open(swagger).read if swagger[0..7] =~ %r{^https?://}
+          if File.exist?(swagger) || swagger[0..7] =~ %r{^https?://}
+            swagger = open(swagger).read
+          end
           begin
             JSON.parse(swagger)
           rescue JSON::ParserError
             begin
               YAML.load(swagger)
             rescue Psych::SyntaxError
               raise 'Only filenames or raw or parsed strings of JSON or YAML are supported.'
-            else
-              Swagger.build(swagger, format: :yaml)
             end
-          else
-            Swagger.build(swagger, format: :json)
           end
         end
       end
 
       def parsed_to_hash(parsed)
         ret = {}
-        parsed.operations.each do |endpoint|
-          ret["#{endpoint.verb} #{endpoint.path.gsub(/{.*?}/, '{}')}"] = endpoint
+        verbs = Set['get', 'put', 'post', 'delete', 'options', 'head', 'patch']
+        parsed['paths'].each do |path, items|
+          # TODO: this doesn't handle external definitions ($ref).
+          warn 'External definitions are not (yet) supported' if items.key?('$ref')
+          (verbs & items.keys).each do |verb|
+            if items['parameters']
```

Comment:
This code looks fine. Stylistically, you could probably extract the parameter rectifying code to its own method.  Also not sure if you care that this code changes the parsed object.